cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -190,6 +190,13 @@ struct Subtract {
   }
 };
 
+struct SubtractTemporal32 {
+  template <typename T, typename Arg0, typename Arg1>
+  static constexpr T Call(KernelContext*, Arg0 left, Arg1 right, Status*) {
+    return static_cast<T>(arrow::internal::SafeSignedSubtract(left, right));
+  }
+};
+
```

Comment:
For some reason [enable_if_time](https://github.com/apache/arrow/blob/5918a160d190164a7b15ef3b0a554ac6b59b0721/cpp/src/arrow/type_traits.h#L703) won't work so I'm adding `SubtractTemporal32`. Would prefer to have this login in `Subtract`. Any ideas why `enable_if_time` won't trigger on `Time32Type`?